ironfish/src/blockchain/blockchain.ts

```diff
@@ -1315,11 +1315,12 @@ export class Blockchain {
   }
 
   private updateSynced(): void {
-    if (this.synced) {
+    if (this.head.timestamp.valueOf() < Date.now() - MAX_SYNCED_AGE_MS) {
+      this.synced = false
```

Comment:
This has very bad implications, and basically will cause non dominant miners to not be able to mine, especially when we have issues with getting gossiped blocks right now. I strongly recommend this as it will cause the network to halt in many cases. If you want something to stop in this case, you probably want to instead check if you're connected to the network.